cadquery/occ_impl/assembly.py

```diff
@@ -290,3 +294,145 @@ def toJSON(
         rv.extend(toJSON(child, loc, color, tolerance))
 
     return rv
+
+
+def toSimplified(assy: AssemblyProtocol, assy_name: str) -> TDocStd_Document:
+    """
+    Converts the assembly to a compound and saves that within the document
+    to be exported.
+
+    :param assy: Assembly that is being converted to a compound for the document.
+    :param assy_name: Label that is applied to the top level object in the document.
+    """
+
+    # Prepare a document
+    app = XCAFApp_Application.GetApplication_s()
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+    app.InitDocument(doc)
+
+    # Shape and color tools
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+    shape_tool.SetAutoNaming_s(False)
+
+    # Set up the compound
+    comp = TopoDS_Compound()
+    comp_builder = BRep_Builder()
+    comp_builder.MakeCompound(comp)
+    for part in assy.children:
+        for solid in part.obj.solids().all():
```

Comment:
I think obj is not always of type Workplane. Could you check with mypy by typing solid first? You might want to use `cq.Assemlby.shapes` here